### DISCUSSION TITLE
Checklist for Building Minimize Size Extension

### DISCUSSION THREAD

**Discussion by johnsoncodehk** [volarjs/volar.js#6] [Show and tell]
- [ ] Bundle JS code with ESBuild using `--minify`
- [ ] Language Client should only import types and shared values from Language Server, to avoid indirect imports of Language Server's runtime logic
- [ ] Check generated `.vsix` and list unneeded files in `.vscodeignore`
- [ ] Load TypeScript from VSCode or user project instead of bundling it
- [ ] Consider avoiding bundling Prettier
- [ ] If Prettier must be bundled, hardcode the exclusion of unwanted parsers
- [ ] Consider replacing the Compiler with a smaller alternative, e.g. `@angular/compiler` -> `@angular-eslint/bundled-angular-compiler`
- [ ] If a full Compiler is required, consider loading the Compiler from user project instead of bundling it
- [ ] Don't include node and browser code in a single `.vsix`, instead separate targets for publishing using `vsce publish --target`
- [ ] Use `esbuild-visualizer` to check your bundle
- [ ] Consider a `Changelog.md` with hyperlinks instead of full content

Unsolved Problems:

- Alternatives to the large `data/webCustomData.js` for `vscode-css-languageservice` and `vscode-html-languageservice`

**Comment by tjx666** [comment]
Recommend my boilerplate: https://github.com/tjx666/awesome-vscode-extension-boilerplate
